sycl/include/CL/sycl/handler.hpp

```diff
@@ -655,33 +431,81 @@ class handler {
-      MArgs.clear();
-      extractArgsAndReqsFromLambda(MHostKernel->getPtr(), KI::getNumParams(),
-                                   &KI::getParamDesc(0));
-      MKernelName = KI::getName();
-      MOSModuleHandle = detail::OSUtil::getOSModuleHandle(KI::getName());
-    } else {
-      // In case w/o the integration header it is necessary to process
-      // accessors from the list(which are associated with this handler) as
-      // arguments.
-      MArgs = std::move(MAssociatedAccesors);
+  /// Requires access to the memory object associated with the placeholder
+  /// accessor.
+  ///
+  /// The command group has a requirement to gain access to the given memory
+  /// object before executing.
+  ///
+  /// @param Acc is a SYCL accessor describing required memory region.
+  template <typename DataT, int Dims, access::mode AccMode,
+            access::target AccTarget>
+  void
+  require(accessor<DataT, Dims, AccMode, AccTarget, access::placeholder::true_t>
+              Acc) {
+    detail::AccessorBaseHost *AccBase = (detail::AccessorBaseHost *)&Acc;
+    detail::AccessorImplPtr AccImpl = detail::getSyclObjImpl(*AccBase);
+    detail::Requirement *Req = AccImpl.get();
+    // Add accessor to the list of requirements.
+    MRequirements.push_back(Req);
+    // Store copy of the accessor.
+    MAccStorage.push_back(std::move(AccImpl));
+    // Add an accessor to the handler list of associated accessors.
+    // For associated accessors index does not means nothing.
+    MAssociatedAccesors.emplace_back(detail::kernel_param_kind_t::kind_accessor,
+                                     Req, static_cast<int>(AccTarget),
+                                     /*index*/ 0);
+  }
+
+  /// Registers event dependencies on this command group.
+  ///
+  /// @param Event is a valid SYCL event to wait on.
+  void depends_on(event Event) {
+    MEvents.push_back(std::move(detail::getSyclObjImpl(Event)));
+  }
+
+  /// Registers event dependencies on this command group.
+  ///
+  /// @param Event is a vector of valid SYCL events to wait on.
+  void depends_on(vector_class<event> Events) {
+    for (event e : Events) {
+      depends_on(e);
     }
   }
 
-  // single_task version with a kernel represented as a lambda.
+  /// Sets argument for OpenCL interoperability kernels.
+  ///
+  /// Registers Arg passed as argument # ArgIndex.
+  ///
+  /// @param ArgIndex is a positional number of argument to be set.
+  /// @param Arg is an argument value to be set.
+  template <typename T> void set_arg(int ArgIndex, T &&Arg) {
+    setArgHelper(ArgIndex, std::move(Arg));
+  }
+
+  /// Sets arguments for OpenCL interoperability kernels.
+  ///
+  /// Registers pack of arguments(Args) with indexes starting from 0.
+  ///
+  /// @param Args are argument values to be set.
+  template <typename... Ts> void set_args(Ts &&... Args) {
+    setArgsHelper(0, std::move(Args)...);
+  }
+
+  /// Defines and invokes a SYCL kernel function as a lambda function
```

Comment:
```suggestion
  /// Defines and invokes a SYCL kernel function as a function object type.
```
Lambda expression is "unnamed function object", so we can simplify "as a lambda function or a named function object type" to "as a function object type".
I suggest applying this to all similar instances in this file.